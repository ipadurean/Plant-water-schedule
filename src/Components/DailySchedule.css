.schedule {
  padding: 2%;
}

.schedule thead th{
  background-color: rgb(187, 206, 104);
  height: 30px;
  width: 250px;
}

.schedule td{
  width: 150px;
  height: 20px;
  background-color: beige;
  text-align: center;
  padding: 3px 0px;
}

#table-top{
  background-color: rgb(211, 190, 97);
}

