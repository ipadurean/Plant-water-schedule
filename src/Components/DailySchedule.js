import React from 'react';
import data from '../plant-data.json';
import './DailySchedule.css'



const DailySchedule = (props) => {
  
  let startingDay = new Date(2019, 11, 16).getTime();
  let plantArr = data.map(el => {return {...el, 'water_after': parseInt(el.water_after), 'starting_point': 1, 'days': [1]}});
  
  let createSchedule = (arr) => {
      let i = 1;
      while (i <= 84){
          for (let el of arr){
              if ((i%7 === 6) && ((i-el.starting_point)% el.water_after === 0)){
                el.days.push(i-1);
                el.starting_point = i-1;
            } else if((i%7 === 0) && ((i-el.starting_point)%el.water_after === 0)) {
                el.days.push(i+1);
                el.starting_point = i+1;
            } else if((i-el.starting_point) && (i-el.starting_point)%el.water_after === 0){
                el.days.push(i);
            }
          }
        i++
      }  
   return arr
  }
  
  let dailySchedule = (selectedDay) => {
    let nthDay = (parseInt(selectedDay) - startingDay)/(1000*60*60*24)+1;
    return createSchedule(plantArr).filter(elem => elem.days.includes(nthDay))
  }
  


  return (
    <div className="schedule">
       
        <table>
          <thead>
            <tr>
              <th colSpan="2" id="table-top"> Plants that need to be watered on: <u>{new Date(parseInt(props.selectedDay)).toDateString()}</u></th>
            </tr>
            <tr>
              <th>Plant name</th>
              <th>Needs water every </th>
            </tr>
          </thead>
          <tbody>{dailySchedule(props.selectedDay).map((el, index) => {
              return <tr key={index}>
                          <td>{el.name}</td><td>{el.water_after} days</td>
                      </tr>
          })}
          </tbody>
        </table>
    </div>
  )

}

export default DailySchedule;