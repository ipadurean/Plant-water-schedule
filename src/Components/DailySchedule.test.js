import React from 'react';
import { shallow } from 'enzyme';
import DailySchedule from './DailySchedule';

describe("DailySchedule", () => {
      
      const app = shallow(<DailySchedule />);

      it ("renders correctly", () => {
        expect(app).toMatchSnapshot();
      });

})