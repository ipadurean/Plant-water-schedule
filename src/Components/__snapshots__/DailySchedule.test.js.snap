// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailySchedule renders correctly 1`] = `ShallowWrapper {}`;
